### PR TITLE
[SYCL] Update sub-group mask test

### PR DESCRIPTION
**Pull request by vladimirlaz**
The test is updated according spec changes in intel/llvm#4481

### CODE REVIEW

**Comment by vladimirlaz** [comment]
@Pennycook, please review/approve the PR?

File: SYCL/SubGroupMask/Basic.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                uint32_t r = 0;
+                gmask_gid2.extract_bits(r);
+                res |= (r != 0xaaaaaa00) << 4;
+                (gmask_gid2 >> 4).extract_bits(r);
+                res |= (r != 0x0aaaaaa0) << 5;
+                gmask_gid3.insert_bits((char)0b01010101, 8);
+                res |= (!gmask_gid3[8] || gmask_gid3[9] || !gmask_gid3[10] ||
+                        gmask_gid3[11])
+                       << 6;
+                marray<unsigned char, 6> mr{1};
```
Comment (Pennycook) [inline SYCL/SubGroupMask/Basic.cpp:65]: Sorry if this is obvious, but why is this a 6?

Comment (vladimirlaz) [reply SYCL/SubGroupMask/Basic.cpp:65]: Just longer-than-sub_group_mask-can accept.

**Comment by vladimirlaz** [comment]
@Pennycook, could you please approve if you are ok?

**Review by Pennycook — APPROVED** [review]

**Pull request merged**